Bump gdbm version to 1.23

FlorentClarret opened the pull request:
Used by `cyrus-sasl.rb` in this PR: https://github.com/DataDog/datadog-agent/pull/15470. 

Bumping this version as it is currently pretty old

Note: 

- We tested the docker image generated in the related PR, everything's OK so far.
- Looks like we are the only one to use this formula

FlorentClarret commented on config/software/gdbm.rb:
```diff
@@ -44,6 +44,7 @@
 
   configure_command = ["./configure",
                        "--enable-libgdbm-compat",
+                       ""--without-readline",
```
To reviewer: added because I was facing this error on osx:

```
  CC       var.o
  CC       util.o
  CC       wordwrap.o
  CC       input-rl.o
input-rl.c:194:12: error: implicit declaration of function 'history_list' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    return history_list ()[n]->line;
           ^
input-rl.c:194:12: note: did you mean 'history_get'?
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/readline/history.h:153:13: note: 'history_get' declared here
HIST_ENTRY      *history_get(int);
                 ^
input-rl.c:194:27: error: subscripted value is not an array, pointer, or vector
    return history_list ()[n]->line;
           ~~~~~~~~~~~~~~~^~
```

I stole the fix from the homebrew formula and it worked well for me: https://github.com/Homebrew/homebrew-core/blob/d48ec91040f4c3deb013d734f89897ad56d4357b/Formula/gdbm.rb

FlorentClarret commented on config/software/gdbm.rb:
```diff
@@ -44,6 +44,7 @@
 
   configure_command = ["./configure",
                        "--enable-libgdbm-compat",
+                       "--without-readline",
```
To reviewer: added because I was facing this error on osx:

```
  CC       var.o
  CC       util.o
  CC       wordwrap.o
  CC       input-rl.o
input-rl.c:194:12: error: implicit declaration of function 'history_list' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    return history_list ()[n]->line;
           ^
input-rl.c:194:12: note: did you mean 'history_get'?
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/readline/history.h:153:13: note: 'history_get' declared here
HIST_ENTRY      *history_get(int);
                 ^
input-rl.c:194:27: error: subscripted value is not an array, pointer, or vector
    return history_list ()[n]->line;
           ~~~~~~~~~~~~~~~^~
```

I stole the fix from the homebrew formula and it worked well for me: https://github.com/Homebrew/homebrew-core/blob/d48ec91040f4c3deb013d734f89897ad56d4357b/Formula/gdbm.rb

bkabrda replied:
I think readline would only be required by the files in `bin` to improve user experience, so I think this is perfectly fine.

bkabrda commented on config/software/gdbm.rb:
```diff
@@ -44,6 +44,7 @@
 
   configure_command = ["./configure",
```
Could we configure with `--disable-static`? We don't want to ship static libraries, so let's ensure explicitly they're not built.

FlorentClarret replied:
Sure

bkabrda reviewed (changes requested):
:wave: hi, this mostly looks good, but I left a comment inline to address.

Could we also leave an explicit block specifying license and license file? Something like the following should do the trick:

```
license "GPL-3.0"
license_file "COPYING"
```

FlorentClarret commented:
> 👋 hi, this mostly looks good, but I left a comment inline to address.
> 
> Could we also leave an explicit block specifying license and license file? Something like the following should do the trick:
> 
> ```
> license "GPL-3.0"
> license_file "COPYING"
> ```

Hey 👋 Thanks for the review. I addressed your comment and request another review a bit later, we need to test the agent image with these changes first

yzhan289 commented:
Closing this PR since we will no longer be using `gdbm` in the agent due to the GPL license issue. We will use `lmdb` instead.
